### DISCUSSION TITLE
Expansion Content Padding

### DISCUSSION THREAD

**Discussion by PoshoDev** [zauberzeug/nicegui#3527] [Q&A] [answered]
### Question

I'm using the `ui.expansion` to create a navigation drawer, but I can't seem to adjust the vertical space between the header and the first item ("Button 1").

```python
from nicegui import ui

with ui.expansion('My Expansion').classes("border").style("padding: 0px; margin: 0px; gap: 0px;"):
    with ui.column().classes("w-full border").style("border-color: red"):
        ui.button('Button 1').classes("w-full")
        ui.button('Button 2').classes("w-full")

ui.run()
```

![2024-08-17_00-39](https://github.com/user-attachments/assets/06481129-b5d2-4863-b916-69992e9aeea6)
I need to adjust the space highlighted in blue.

I've also tried setting padding and margin to `0` with `ui.add_css()` on `.q-expansion-item__content`, `.q-item`, `.q-expansion-item__header`, and other related classes, but I can't get it to work.

None of these seem to affect the vertical spacing. How can I eliminate the gap between the header and the first item? Could anyone offer some advice on how to get it to work? Thanks a lot!

**Comment by python-and-novella** [comment, marked as answer]
```python3
with ui.expansion('expansion').classes('bg-grey'):
    ui.query('.nicegui-expansion .q-expansion-item__content').style('padding:0',replace='gap:0')
    ui.button('button').classes('w-full h-full')
    ui.button('button').classes('w-full h-full')
```
![image](https://github.com/user-attachments/assets/58743a6b-9a64-4edb-8e9d-4c7e77c4955f)

**Reply by PoshoDev** [reply]
> ```python
> with ui.expansion('expansion').classes('bg-grey'):
>     ui.query('.nicegui-expansion .q-expansion-item__content').style('padding:0',replace='gap:0')
>     ui.button('button').classes('w-full h-full')
>     ui.button('button').classes('w-full h-full')
> ```

Thank you so much! I definitely would've never gotten to that solution on my own!

One slight thing to note about this solution would be that every subsequent expansion would have the same `padding` and `gap` after running the `ui.query`:
```python
from nicegui import ui

with ui.expansion('expansion').classes('bg-grey'):
    ui.query('.nicegui-expansion .q-expansion-item__content').style('padding:0',replace='gap:0')
    ui.button('button').classes('w-full h-full')
    ui.button('button').classes('w-full h-full')

with ui.expansion('My Expansion').classes("border").style("padding: 0px; margin: 0px; gap: 0px;"):
    with ui.column().classes("w-full border").style("border-color: red"):
        ui.button('Button 1').classes("w-full")
        ui.button('Button 2').classes("w-full")

ui.run()
```
![image](https://github.com/user-attachments/assets/9b623029-54ca-4d41-bfcf-44e800d58886)

However, that's something I can work with.

Thank you so much for the help! :)

**Reply by python-and-novella** [reply]
you can select id like this:
```python3
with ui.expansion('expansion') as exp1:
    ui.query(f'#c{exp1.id} .q-expansion-item__content').style('padding:0',replace='gap:0')
    ui.button('button').classes('w-full h-full')
    ui.button('button').classes('w-full h-full')

with ui.expansion('expansion'):
    ui.button('button').classes('w-full h-full')
    ui.button('button').classes('w-full h-full')
```
![image](https://github.com/user-attachments/assets/8064e53c-6f02-44ad-b788-233ddb660220)

**Reply by PoshoDev** [reply]
>     ui.query(f'#c{exp1.id} .q-expansion-item__content').style('padding:0',replace='gap:0')
That did the trick so well! Thank you so much! 😁
